packages/odoo/test/Adaptor.test.js

```diff
@@ -178,3 +180,114 @@ describe('delete', () => {
     expect(data).to.eql(true);
   });
 });
+
+describe('searchReadRecord', () => {
+  const makeData = n => {
+    const rows = [];
+    for (let i = 0; i < n; i++) {
+      rows.push({
+        id: i + 1,
+        name: `P${i + 1}`,
+        is_company: true,
+        im_status: 'im_partner',
+      });
+    }
+    rows.push({
+      id: 10001,
+      name: 'NonMatch',
+      is_company: false,
+      im_status: 'im_partner',
+    });
+    rows.push({
+      id: 10002,
+      name: 'VendorOnly',
+      is_company: true,
+      im_status: 'im_vendor',
+    });
+    return rows;
+  };
+
+  it('fetches up to the limit using multiple pages and returns nextOffset=null at the end', async () => {
```

Comment:
Good - but I disagree with the language of  `fetches up to the limit` here. It fetches all items, but only because the default limit is lower than the number of items.